### PR TITLE
[image-picker] Add support for video recording via `launchCameraAsync`

### LINKED ISSUE #4310: Verify on Android whether "mediaTypes" is ignored when launching a camera instead of library picker
> We have _Take photo_/_Take video_ in ImagePicker examples in NCL, but looks like setting `mediaTypes` up on iOS was intentionally skipped when launching a camera instead of the image library picker. Such option is also not documented for `launchCameraAsync`.

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Resolve #4310. 

# How

Added `mediaType` support for Android. 

# Test Plan

You can check it on https://snack.expo.io/@lukaszkosmaty/874a3d

### CODE REVIEW

File: packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m [hunk 1]
```diff
@@ -294,16 +294,17 @@ - (BOOL)tryCopyImage:(NSDictionary * _Nonnull)info path:(NSString *)toPath {
...
   NSURL *videoURL = info[UIImagePickerControllerMediaURL];
-  PHFetchResult<PHAsset *> *assets = [PHAsset fetchAssetsWithALAssetURLs:@[[info valueForKey:UIImagePickerControllerReferenceURL]] options:nil];
-  if (assets.count > 0) {
-    PHAsset *videoAsset = assets.firstObject;
-    response[@"width"] = @(videoAsset.pixelWidth);
-    response[@"height"] = @(videoAsset.pixelHeight);
-    response[@"duration"] = @(videoAsset.duration * 1000);
-  } else {
-    UMLogInfo(@"Could not fetch metadata for video %@", [videoURL absoluteString]);
+  if (info[UIImagePickerControllerReferenceURL]){ // video from gallery
```
Comment (tsapeta) [inline packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m]:
```suggestion
  if (info[UIImagePickerControllerReferenceURL]) { // video from gallery
```

File: packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m [hunk 2]
```diff
@@ -328,6 +329,18 @@ - (void)handleVideoWithInfo:(NSDictionary * _Nonnull)info saveAt:(NSString *)dir
 
   NSURL *fileURL = [NSURL fileURLWithPath:path];
   NSString *filePath = [fileURL absoluteString];
+  
+  // adding data to response if video came from camera
+  if (!info[UIImagePickerControllerReferenceURL]){
```
Comment (tsapeta) [inline packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m]:
```suggestion
  if (!info[UIImagePickerControllerReferenceURL]) {
```

File: docs/pages/versions/unversioned/sdk/imagepicker.md [hunk 3]
```diff
@@ -56,6 +56,7 @@ Display the system UI for taking a photo with the camera. Requires `Permissions.
 
   A map of options:
 
+  - **mediaTypes (_String_)** -- Choose what type of media to pick. Usage: `ImagePicker.MediaTypeOptions.<Type>`, where `<Type>` is one of: `Images`, `Videos`, `All` (only on iOS). Defaults to `false`
```
Comment (tsapeta) [inline docs/pages/versions/unversioned/sdk/imagepicker.md]: Doesn't it default to `Images`?

File: CHANGELOG.md [hunk 4]
```diff
@@ -22,6 +22,7 @@ This is the log of notable changes to the Expo client that are developer-facing.
 
 - Added `fallbackLabel` option to `LocalAuthentication.authenticateAsync` on iOS which allows to customize a title of the fallback button when the system doesn't recognize the user and asks to authenticate via device passcode. ([#4612](https://github.com/expo/expo/pull/4612) by [@changLiuUNSW](https://github.com/changLiuUNSW))
 - added `native` mode for Android SplashScreen on standalone apps by [@bbarthec](https://github.com/bbarthec) ([#4567](https://github.com/expo/expo/pull/4567))
+- added support for video recording in `ImagePicker.launchCameraAsync` by [@lukmccall](https://github.com/lukmccall)
```
Comment (tsapeta) [inline CHANGELOG.md]:
```suggestion
- added support for video recording in `ImagePicker.launchCameraAsync`. ([#4903](https://github.com/expo/expo/pull/4903) by [@lukmccall](https://github.com/lukmccall))
```

We're changing the format of changelog entries 😉

**Review by tsapeta — CHANGES_REQUESTED** [review]

**Review by tsapeta — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -22,6 +22,7 @@ This is the log of notable changes to the Expo client that are developer-facing.
 
 - Added `fallbackLabel` option to `LocalAuthentication.authenticateAsync` on iOS which allows to customize a title of the fallback button when the system doesn't recognize the user and asks to authenticate via device passcode. ([#4612](https://github.com/expo/expo/pull/4612) by [@changLiuUNSW](https://github.com/changLiuUNSW))
 - added `native` mode for Android SplashScreen on standalone apps by [@bbarthec](https://github.com/bbarthec) ([#4567](https://github.com/expo/expo/pull/4567))
+- added support for video recording in `ImagePicker.launchCameraAsync`. ([#4903](https://github.com/expo/expo/pull/4903) by [@lukmccall](https://github.com/lukmccall))
```
Comment (bbarthec) [inline CHANGELOG.md]: nitpicking, by convention is to first describing `by [@who]` part and then PR reference 😉

File: packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.java [hunk 6]
```diff
@@ -255,7 +257,9 @@ public void launchCameraAsync(final Map<String, Object> options, final Promise p
       return;
     }
 
-    final Intent cameraIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+    final Intent cameraIntent = new Intent(mediaTypes.equals("Videos") ?
+                                            MediaStore.ACTION_VIDEO_CAPTURE :
+                                            MediaStore.ACTION_IMAGE_CAPTURE);
```
Comment (bbarthec) [inline packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.java]: Is indentation here in order? 🤔 Dunno, do not bother 😝

**Review by bbarthec — APPROVED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (tsapeta) [reply CHANGELOG.md]: You're not right @bbarthec 😛 That's how we did, but now I'm encouraging to use `(#4903 by @lukmccall) instead. This way it is more clear and the fact who did the change isn't as important as the link to the pull request where people can find more info.

**Pull request merged**